### ISSUE TITLE
provider/aws: Use Terraform User-Agent for API requests

### ISSUE THREAD

**Issue by radeksimko** [hashicorp/terraform#5020]
This would be helpful for tracking down API requests and/or even doing some stats from CloudTrail data.

Here's an example of current CloudTrail event coming from Terraform:

```json
{
    "eventVersion": "1.03",
    "userIdentity": {
        "type": "IAMUser",
        "principalId": "AAAAAAAAAAAAAAAA",
        "arn": "arn:aws:iam::111111111111:user/username",
        "accountId": "111111111111",
        "userName": "username"
    },
    "eventTime": "2016-02-05T18:37:55Z",
    "eventSource": "s3.amazonaws.com",
    "eventName": "CreateBucket",
    "awsRegion": "us-east-1",
    "sourceIPAddress": "8.8.8.8",
    "userAgent": "[aws-sdk-go/1.1.0 (go1.5.2; darwin; amd64)]",
    "requestParameters": {
        "x-amz-acl": [
            "private"
        ],
        "bucketName": "my-own-bucket"
    },
    "responseElements": null,
    "requestID": "*REDACTED*",
    "eventID": "*REDACTED*",
    "eventType": "AwsApiCall",
    "recipientAccountId": "111111111111"
}
```

I believe we should replace `aws-sdk-go/1.1.0 (go1.5.2; darwin; amd64)` with Terraform version. I don't mind keeping the AWS-SDK version there, but I think that as long as users use released versions of Terraform, it should be implicitly possible to track down version of the SDK from the Terraform version.

**Comment by stack72** [comment]
I think this is a great idea :)

**Comment by kapilt** [comment]
looks like its just a matter of setting the variable / alternatively setting up a named handler on the session

session construction with
```golang
session.handlers.Build.PushBackNamed(TerraformAgentHandler)

var TerraformAgentHandler = request.NamedHandler{
        Name: "core.SDKVersionUserAgentHandler",
        Fn: request.MakeAddToUserAgentHandler("terraform", "0.6.3")
}
```

**Comment by phinze** [comment]
Done and merged in #5621 :+1:
